client/benches/tps/lib.rs

```diff
@@ -203,7 +203,9 @@ impl MeasurerUnit {
         thread::spawn(move || -> Result<()> {
             for instruction in instructions {
                 submitter.submit(instruction)?;
-                thread::sleep(core::time::Duration::from_micros(interval_us_per_tx));
+                if interval_us_per_tx > 0 {
+                    thread::sleep(core::time::Duration::from_micros(interval_us_per_tx));
+                }
```

Comment:
I wonder if both this and `thread::sleep(0_duration)` optimises to the same thing.

Reply:
Nope. https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,selection:(endColumn:10,endLineNumber:3,positionColumn:10,positionLineNumber:3,selectionStartColumn:10,selectionStartLineNumber:3,startColumn:10,startLineNumber:3),source:'//+Type+your+code+here,+or+load+an+example.%0Apub+fn+sleep_if(num:+u64)+-%3E+i32+%7B%0A++++if+num+%3E+0+%7B%0A++++++++std::thread::sleep(core::time::Duration::from_micros(num))%0A++++%7D%3B%0A++++0%0A%7D%0A%0A//+Type+your+code+here,+or+load+an+example.%0Apub+fn+sleep_no_if(num:+u64)+-%3E+i32+%7B%0A++++std::thread::sleep(core::time::Duration::from_micros(num))%3B%0A++++0%0A%7D%0A%0A//+If+you+use+%60main()%60,+declare+it+as+%60pub%60+to+see+it+in+the+output:%0A//+pub+fn+main()+%7B+...+%7D%0A'),l:'5',n:'0',o:'Rust+source+%231',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:r1640,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:rust,libs:!(),options:'-O',selection:(endColumn:1,endLineNumber:15,positionColumn:1,positionLineNumber:15,selectionStartColumn:1,selectionStartLineNumber:14,startColumn:1,startLineNumber:14),source:1,tree:'1'),l:'5',n:'0',o:'rustc+1.64.0+(Rust,+Editor+%231,+Compiler+%231)',t:'0')),k:33.333333333333336,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'rustc+1.64.0',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+rustc+1.64.0+(Compiler+%231)',t:'0')),k:33.33333333333333,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4

Reply:
Please remove the conditional. As per [the official docs ](https://doc.rust-lang.org/std/thread/fn.sleep.html), the behaviour is exactly the same, and the call never gets inlined. 

> Currently, specifying a zero duration on Unix platforms returns immediately without invoking the underlying [nanosleep](https://linux.die.net/man/2/nanosleep) syscall, whereas on Windows platforms the underlying [Sleep](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-sleep) syscall is always invoked. If the intention is to yield the current time-slice you may want to use [yield_now](https://doc.rust-lang.org/std/thread/fn.yield_now.html) instead.

Reply:
So I originally put the if there because of some benchmarks I ran. I was now unable to replicate them so I will remove the conditional.